json/shared/src/main/scala/fs2/data/json/JsonContext.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2
+package data
+package json
+
+import cats.Show
+import cats.implicits._
+
+/** A JSON trace context, positioning the current location in the JSON value
+  * starting from the root.
+  */
+sealed trait JsonContext
+
+object JsonContext {
+  case object Root extends JsonContext
+  case class Key(name: String, parent: JsonContext) extends JsonContext
+  case class Index(idx: Long, parent: JsonContext) extends JsonContext
+
+  implicit object JsonContextShow extends Show[JsonContext] {
```

Comment:
I find it cool to have `Show` instances for all these and I'm wondering whether it's worth to have instances of `Eq` too?

Reply:
I want this to be used for reporting purpose, I am not convinced whether `Eq` is important in such use cases.